pkg/csi/service/vanilla/controller.go

```diff
@@ -941,13 +941,24 @@ func (c *controller) ControllerExpandVolume(ctx context.Context, req *csi.Contro
 	log := logger.GetLogger(ctx)
 	log.Infof("ControllerExpandVolume: called with args %+v", *req)
 
+	isExtendSupported, err := c.manager.VcenterManager.IsExtendVolumeSupported(ctx, c.manager.VcenterConfig.Host)
+	if err != nil {
+		log.Errorf("failed to verify if extend volume is supported or not. Error: %+v", err)
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	if !isExtendSupported {
+		msg := "Volume Expansion is not supported in this vSphere release. Kindly upgrade to vSphere 7.0 for offline expansion and vSphere 7.0U2 for online expansion support."
+		log.Error(msg)
+		return nil, status.Error(codes.Internal, msg)
+	}
+
 	if strings.Contains(req.VolumeId, ".vmdk") {
```

Comment:
Can we move this `if strings.Contains(req.VolumeId, ".vmdk") {` before `isExtendSupported` check?

Reply:
Done